Source/Plugin.BLE.Android/Adapter.cs

```diff
@@ -184,6 +185,34 @@ public override List<IDevice> GetSystemConnectedOrPairedDevices(Guid[] services
             return connectedDevices.Union(bondedDevices, new DeviceComparer()).Select(d => new Device(this, d, null, 0)).Cast<IDevice>().ToList();
         }
 
+        protected override async Task<IDevice> ConnectNativeAsync(string friendlyName, Guid uuid, Func<IDevice, bool> deviceFilter, CancellationToken cancellationToken = default)
+        {
+            if (uuid == Guid.Empty)
+            {
+                var taskCompletionSource = new TaskCompletionSource<IDevice>();
+                EventHandler<DeviceEventArgs> handler = (sender, args) =>
+                {
+                    taskCompletionSource.TrySetResult(args.Device);
+                };
+
+                DeviceDiscovered += handler;
+
+                async Task<IDevice> WaitAsync()
```

Comment:
Inner tasks - I prefer having this outside, but it is exciting 🐙

Reply:
It is super exiting 😄. I think inner method is fine, since this method shouldn't be called outside of the outer method.